synapse/membrane.py

```diff
@@ -0,0 +1,169 @@
+# FIXME support ival splices
+# FIXME implement roles/groups/users/perms
+
+import copy
+import fnmatch
+import operator
+
+from synapse.eventbus import EventBus
+import synapse.lib.syntax as s_syntax
+
+class Membrane(EventBus):
+
+    def __init__(self, src=None, dst=None, rules=None, default=None):
+        '''
+        Filters Cortex splices based on rules and forwards to destination Cortex if rules allow.
+
+        Args:
+            dst: the destination Cortex to forward messages to
+            rules: the filter rules
+        '''
+        EventBus.__init__(self)
+
+        _SPLICE_NAMES = (
+            'node:add',
+            'node:del',
+            'node:prop:set',
+            'node:prop:del',
+            'node:tag:add',
+            'node:tag:del',
+            #'node:ival:set',
+            #'node:ival:del',
+        )
+        _DEFAULT_RULE = [{'query': ''}]
+        _DEFAULT_RULES = {'': {k: copy.deepcopy(_DEFAULT_RULE) for k in _SPLICE_NAMES}}
+
+        self.rules = copy.deepcopy(_DEFAULT_RULES)
+        if rules:
+            for key in rules:
+                for rule in rules[key]:
+                    if self._is_valid_rule(rule):
+                        self.rules[''][key].append(rule)
+
+        self.opers = {
+            'eq': operator.eq,
+            'ne': operator.ne,
+            'le': operator.le,
+            'ge': operator.ge,
+            'lt': operator.lt,
+            'gt': operator.gt,
+
+            'has': operator.eq,
```

Comment:
`has` is what `s_syntax` call the oper it uses when doing a filter on a prop w/o a valu